src/java_bytecode/gather_methods_lazily.cpp

```diff
@@ -0,0 +1,499 @@
+/*******************************************************************\
+
+ Module: Java Bytecode
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+#include "gather_methods_lazily.h"
+
+
+#include <java_bytecode/java_entry_point.h>
+#include <java_bytecode/java_class_loader.h>
+#include <java_bytecode/java_utils.h>
+#include <util/safe_pointer.h>
+#include <util/suffix.h>
+#include <java_bytecode/java_string_library_preprocess.h>
+
+
+gather_methods_lazilyt::gather_methods_lazilyt(
```

Comment:
Suggest keeping the `ci_lazy_methods` name (CI, context-insensitive, will differentiate this from future smarter ways of lazily faulting methods in). Alternatively, rename e.g. the constants in `java_bytecode_languaget` and the documentation to be consistent with any renaming.